### PR TITLE
Support joining game

### PR DESCRIPTION
**Pull request by Nicholaswithak**
1.Added link <%= link_to 'Join This Game', game_path(@game), method: :patch %> to show.html.erb

2.added a Show method in -> gamescontroller.rb

I worked on this with my mentor, it was brought up that this link wont work if there isnt' a game in your database, and the join game wont work unless there are users that are in a current game.

### CODE REVIEW

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -11,14 +11,19 @@ def create
...
   end
 
-  def show; end
+  def show
+    @game = Game.find(params[:id])
+  end
 
   def update
     @game = Game.find_by_id(:id)
+    @game.update_attributes(black_player_id: current_user.id)
```
Comment (jmoglesby) [inline app/controllers/games_controller.rb]: I think we are going to have to add some conditionals in our `game#update` method here to ensure only the right type of update is happening. As this is written, in combination with our piece movement logic, anytime a user, any user, moves a piece, and we call `game#update`, the `black_player_id` for the game will be set to the user that is moving the piece (even if it's the white player).

File: app/views/games/show.html.erb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 <h1>Games#show</h1>
+<%= link_to 'Join This Game', game_path(@game), method: :patch %>
```
Comment (jmoglesby) [inline app/views/games/show.html.erb]: We should add logic here such that the link is only shown on the page **if** `black_player_id = nil`. We don't want to show this link to people if the game already has both players.

**Comment by jmoglesby** [comment]
The CircleCI failure here should get a pass. Another case where @Nicholaswithak branched off master before some specific testing changes were made, which I think are being applied to his code that has no knowledge of those changes:
image

File: app/views/games/show.html.erb (hunk 2 above)
Comment (Nicholaswithak) [reply app/views/games/show.html.erb]: I didnt make changes to homecontroller.rb or application.html.erb 

I accidentally put a link in application.html.erb but took it out and moved it to show.html.erb

Comment (Nicholaswithak) [reply app/views/games/show.html.erb]: The part of the black_player_id logic is the part I was stuck on and needed your help.

Comment (Nicholaswithak) [reply app/views/games/show.html.erb]: I didnt make changes to homecontroller.rb or application.html.erb 

I accidentally put a link in application.html.erb but took it out and moved it to show.html.erb

**Comment by Nicholaswithak** [comment]
I am getting an IAT error, was new google credentials generated? if so could you guys please send me the new auth. 

thanks.

**Comment by seansinger22** [comment]
> I am getting an IAT error, was new google credentials generated? if so could you guys please send me the new auth.
> 
> thanks.

I put the application.yml file back in slack if you need it again.  Maybe it corrupted somehow?  Authorities have not changed.

**Comment by seansinger22** [comment]
Nicholas is fixing this feature and will push under a new branch.

**Pull request closed without merging**